doc/stdenv/platform-notes.chapter.md

```diff
@@ -2,21 +2,22 @@
 
 ## Darwin (macOS) {#sec-darwin}
 
-Some common issues when packaging software for Darwin:
+The Darwin `stdenv` differs from most other ones in nixpkgs in a few key ways. These differences reflect the default assumptions for building software on that platform. In many cases, you can ignore these differences because the software you are packaging is already written with them in mind. When you do that, write your derivation as normal. You don’t have to include any Darwin-specific special cases. The easiest way to know whether your derivation requires special handling for Darwin is to write it as if it doesn’t and see if it works. If it does, you’re done; skip the rest of this.
 
-- The Darwin `stdenv` uses clang instead of gcc. When referring to the compiler `$CC` or `cc` will work in both cases. Some builds hardcode gcc/g++ in their build scripts, that can usually be fixed with using something like `makeFlags = [ "CC=cc" ];` or by patching the build scripts.
+- Darwin uses clang by default instead of GCC. Packages that refer to `$CC` or `cc` should just work in most cases. Some packages may hardcode `gcc` or `g++`. You can usually fix that by setting `makeFlags = [ "CC=cc" "CXX=C++" ]`. If that does not work, you will have to patch the build scripts yourself to use the correct compiler for Darwin.
+- Darwin needs an SDK to build software. The SDK provides a default set of frameworks and libraries to build software, most of which are specific to Darwin. There are multiple versions of the SDK packages in nixpkgs, but one is included by default in the `stdenv`. Usually, you don’t have to change or pick a different SDK. When in doubt, use the default.
+- The SDK used by your build can be found using the `DEVELOPER_DIR` environment variable. There are also versions of this variable available when cross-compiling depending on the SDK’s role. The `SDKROOT` variable is also set with the path to the SDK’s libraries and frameworks. `SDKROOT` is always a sub-folder of `DEVELOPER_DIR`.
```

Comment:
This can go later in documentation of internals, most users won’t have to know about this and if they run into a situation where they do they’ll need more information than this.

Reply:
I want the opening section to explain what the SDK is and what it contains. If they run into a situation that’s not easily resolvable with basic knowledge, that’s what the rest of the documentation is for.